android-kotlin/app/src/main/java/com/mongodb/app/TaskActivity.kt

```diff
@@ -0,0 +1,136 @@
+package com.mongodb.app
+
+import android.app.AlertDialog
+import android.content.Intent
+import android.os.Bundle
+import android.util.Log
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import android.view.ViewGroup
+import android.widget.EditText
+import android.widget.TextView
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.widget.Toolbar
+import androidx.recyclerview.widget.DividerItemDecoration
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import io.realm.Realm
+import io.realm.RealmConfiguration
+import io.realm.kotlin.where
+import io.realm.mongodb.sync.SyncConfiguration
+
+class TaskActivity : AppCompatActivity() {
+    private lateinit var userRealm: Realm
+    private lateinit var config: RealmConfiguration
+    private lateinit var recyclerView: RecyclerView
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_task)
+
+        val toolbar = findViewById<View>(R.id.task_menu) as Toolbar
+        setSupportActionBar(toolbar)
+
+        val fab = findViewById<View>(R.id.floating_action_button)
+        fab.setOnClickListener { (onFabClicked()) }
+
+        recyclerView = findViewById(R.id.task_list)
+        recyclerView.layoutManager = LinearLayoutManager(this)
+        recyclerView.setHasFixedSize(true)
+        recyclerView.addItemDecoration(DividerItemDecoration(this, DividerItemDecoration.VERTICAL))
+    }
+
+    /**
+     *  On start, open a realm with a partition ID equal to the user ID.
+     *  Query the realm for Task objects, sorted by a stable order that remains
+     *  consistent between runs, and display the collection using a recyclerView.
+     */
+    override fun onStart() {
+        super.onStart()
+        val user = realmApp.currentUser()
+        if (user == null) {
+            startActivity(Intent(this, LoginActivity::class.java))
+        }
+        else {
+            val partition = user.id
+            config = SyncConfiguration.Builder(user, partition).build()
+            Realm.getInstanceAsync(config, object : Realm.Callback() {
+                override fun onSuccess(realm: Realm) {
+                    this@TaskActivity.userRealm = realm
+                    recyclerView.adapter = TaskAdapter(realm.where<Task>().sort("_id").findAll(), config)
```

Comment:
I would use findAllAsync() here, otherwise the query runs on the UI thread which can slow it down.